onedal/datatypes/data_conversion_sua_iface.cpp

```diff
@@ -64,6 +67,23 @@ dal::table convert_to_homogen_impl(py::object obj) {
     // Get oneDAL Homogen DataLayout enumeration from input object shape and strides.
     const auto layout = get_sua_iface_layout(sua_iface_dict, r_count, c_count);
 
+    if (layout == dal::data_layout::unknown){
+        py::object copy;
+        if (py::hasattr(obj, "copy")){
+            copy = obj.attr("copy")();
+        }
+        else if (py::hasattr(obj, "__array_namespace__")){
+            const auto space = obj.attr("__array_namespace__")();
+            copy = space.attr("asarray")(obj, "copy"_a = true);
+        }
+        else {
+            throw std::runtime_error("Wrong strides");
+        }
+        res = convert_to_homogen_impl<Type>(copy);
+        copy.dec_ref();
+        return res;
+    }
```

Comment:
Please add comments here for other reviewers

Reply:
done!